test/extended/builds/docker_pullsecret.go

```diff
@@ -35,29 +35,16 @@ var _ = g.Describe("[builds][pullsecret][Conformance] docker build using a pull
 			err := oc.Run("create").Args("-f", buildFixture).Execute()
 			o.Expect(err).NotTo(o.HaveOccurred())
 
-			g.By("starting a test build")
-			out, err := oc.Run("start-build").Args("docker-build").Output()
-			fmt.Fprintf(g.GinkgoWriter, "\nstart-build output:\n%s\n", out)
-			o.Expect(err).NotTo(o.HaveOccurred())
-
-			g.By("expecting the build succeeds")
-			err = exutil.WaitForABuild(oc.REST().Builds(oc.Namespace()), "docker-build-1", exutil.CheckBuildSuccessFn, exutil.CheckBuildFailedFn)
-			if err != nil {
-				exutil.DumpBuildLogs("docker-build", oc)
-			}
+			g.By("starting a build")
+			br, err := exutil.RunBuildConfigAndWait(oc, "docker-build")
```

Comment:
@bparees 

> i'd suggest keeping the "StartBuild" function name and repurposing it.
Name simplification sounds good. 

> and should StartBuild become StartBuildAndWait ? are there cases where we don't want to wait?
Yep. There are few extended tests that specifically test parallel builds. 

> Also RunStartBuildConfigAndWait can actually take a build argument, no? so it's not well named.
Agreed. 

> 1) start a build and not wait (possibly, needed for the serialized build tests I believe)
My RunStartBuild was meant for this -- advanced tests that can't follow the typical pattern / don't want to pass -o=name  / can't wait for the result.

> 2) start a build and expect failure (so asserting success is not always valid).
There is a AssertFailure() as well. 
https://github.com/jupierce/origin/blob/b9a0173257e99f1767981934793fbc27ba8a1d70/test/extended/builds/docker_quota.go#L42